### PR TITLE
chore: remove unnecessary files

### PR DESCRIPTION
**Pull request by fgdorais**
These files are not necessary since `Std.lean` imports everything.

### CODE REVIEW

**Comment by digama0** [comment]
These files exist so that people can import parts of Std without importing the whole thing. This is an intended use-case. They also exist so that Std can be backward-compatible when splitting files.

File: Std/Lean/Meta/UnusedNames.lean [hunk 1]
```diff
@@ -5,7 +5,7 @@ Authors: Jannis Limperg
 -/
 
 import Lean.Meta
-import Std.Data.String
+import Std.Data.String.Basic
```
Comment (digama0) [inline Std/Lean/Meta/UnusedNames.lean:8]: (that said, within Std itself we should be using minimal imports, so this could go in a separate PR)

**Comment by fgdorais** [comment]
I submitted preemptive patches [aesop#86](https://github.com/leanprover-community/aesop/pull/86) and [mathlib#8538](https://github.com/leanprover-community/mathlib4/pull/8538). Once these are applied, this PR can be merged with no downstream effects to Aesop and Mathlib.
